### PR TITLE
 Fix GDRCopy and re-release PT 1.13 Training DLC

### PR DESCRIPTION
**Pull request by sirutBuasai**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Description
PT 1.13 DLC currently does not support GDRCopy. This PR fixes this issue and re-release PT 1.13 DLC with functioning GDRCopy.

### Tests run
- [e63f769](https://github.com/aws/deep-learning-containers/pull/4030/commits/e63f76902e3174db3c34fa2774b916ea64a6af78) PT 1.13 EC2
  - Passed ec2, ecs, eks, sanity, ec2 benchmark tests.
- [388888f](https://github.com/aws/deep-learning-containers/pull/4030/commits/388888fb6646aa1d5ef4b27a9414549459a0c28b) PT 1.13 SM
  - Passed ec2, ecs, eks, sanity, ec2 benchmark, sm remote, sm local, sm rc, sm efa tests

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [ ] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**
<details>
<summary>Expand</summary>

- [ ] `sagemaker_remote_tests = true`
- [ ] `sagemaker_efa_tests = true`
- [ ] `sagemaker_rc_tests = true`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

</details>

### Formatting
- [ ] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

#### Builds to Execute
<details>
<summary>Expand</summary>

Fill out the template and click the checkbox of the builds you'd like to execute

*Note: Replace with <X.Y> with the major.minor framework version (i.e. 2.2) you would like to start.*

- [x] build_pytorch_training_1.13_sm
- [x] build_pytorch_training_1.13_ec2

- [ ] build_pytorch_inference_<X.Y>_sm
- [ ] build_pytorch_inference_<X.Y>_ec2
- [ ] build_pytorch_inference_<X.Y>_graviton

- [ ] build_tensorflow_training_<X.Y>_sm
- [ ] build_tensorflow_training_<X.Y>_ec2

- [ ] build_tensorflow_inference_<X.Y>_sm
- [ ] build_tensorflow_inference_<X.Y>_ec2
- [ ] build_tensorflow_inference_<X.Y>_graviton
</details>

### PR Checklist 
<details>
<summary>Expand</summary>

- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

#### NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ec2_benchmark_tests = true` or `sagemaker_benchmark_tests = true`
</details>

### Pytest Marker Checklist
<details>
<summary>Expand</summary>

- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type
</details>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/training/docker/1.13/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -111,14 +111,15 @@ RUN rm -rf /root/.ssh/ \
...
 ENV REQUESTS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
 RUN curl -L -o ~/mambaforge.sh https://github.com/conda-forge/miniforge/releases/download/${MAMBA_VERSION}/Mambaforge-${MAMBA_VERSION}-Linux-x86_64.sh \
  && chmod +x ~/mambaforge.sh \
  && ~/mambaforge.sh -b -p /opt/conda \
- && rm ~/mambaforge.sh \
- && /opt/conda/bin/conda update -y conda \
- && /opt/conda/bin/conda install -y -c conda-forge \
+ && rm ~/mambaforge.sh
+
+RUN /opt/conda/bin/mamba install -y -c conda-forge \
```
[the remark this answers is not in the document]
Comment (sirutBuasai) [reply pytorch/training/docker/1.13/py3/Dockerfile.cpu:121]: This helps with layering the image at `docker pull` time. Doesn't matter as much but all PT 2.* separates this so I figured we can just do it in PT 1.13 too.

File: test/dlc_tests/container_tests/bin/pytorch_tests/installOSSPyTorch [hunk 2]
```diff
@@ -9,13 +9,18 @@ PT_VERSION=$(python -c "import torch; print(torch.__version__.split('+')[0]);")
 HAS_CUDA=$(python -c "import torch; print(torch.version.cuda is not None)")
 
 # install OSS PyTorch
-conda remove -y --force pytorch torchvision torchaudio 
+if [ $(version $PT_VERSION) -lt $(version "2.0") ]; then
+    pip uninstall -y torch torchvision torchaudio torchdata
```
[the remark this answers is not in the document]
Comment (sirutBuasai) [reply test/dlc_tests/container_tests/bin/pytorch_tests/installOSSPyTorch:13]: PT 1.* are using pip binaries. These will have to be uninstalled using pip. PT 2.* are using conda binaries, these will have to be installed using conda.

Of course, there could be a case in the future that PT 2.* at some version, we might move back to pip. I tried writing a script that automatically detects whether the torch binary is pip installed or conda installed and remove them accordingly but the script didn't work as expected so I'm doing a version pin solution for now as this is more deterministic.

File: test/dlc_tests/ec2/pytorch/training/common_cases.py [hunk 3]
```diff
@@ -69,7 +69,7 @@ def pytorch_standalone(pytorch_training, ec2_connection):
     )
 
 
-def pytorch_train_mnist(pytorch_training, ec2_connection):
+def pytorch_training_mnist(pytorch_training, ec2_connection):
```
[the remark this answers is not in the document]
Comment (sirutBuasai) [reply test/dlc_tests/ec2/pytorch/training/common_cases.py:72]: To keep with the convention other tests with the naming convention of `pytorch_training_<test_name>`. Eg: https://github.com/aws/deep-learning-containers/blob/71d4d3146405c44d78183f3639d8fe4a524adffa/test/dlc_tests/ec2/pytorch/training/common_cases.py#L112-L140

File: test/dlc_tests/container_tests/bin/pytorch_tests/installOSSPyTorch (hunk 2 above)
[the remark this answers is not in the document]
Comment (sirutBuasai) [reply test/dlc_tests/container_tests/bin/pytorch_tests/installOSSPyTorch:13]: I'm not sure why this has been currently working within the pipeline given that the script has been doing conda uninstalled a pip binary which should not be possible. So pinning this is definitely necessary.

**Review by shantanutrip — APPROVED** [review]

**Pull request merged**